workers/common/src/main/kotlin/common/Utils.kt

```diff
@@ -36,11 +42,15 @@ fun enableOrtStackTraces() {
  * available for a provenance, it is ignored and not included in the result.
  */
 internal fun getFileLists(fileListResolver: FileListResolver, provenances: Set<KnownProvenance>) =
-    provenances
-        .mapNotNull { provenance -> fileListResolver.get(provenance)?.let { provenance to it } }
-        .mapTo(mutableSetOf()) { (provenance, fileList) ->
-            FileList(
-                provenance,
-                fileList.files.mapTo(mutableSetOf()) { FileList.Entry(it.path, it.sha1) }
-            )
+    runBlocking(Dispatchers.IO.limitedParallelism(20)) {
+        provenances.map { provenance ->
+            async { provenance to fileListResolver.get(provenance) }
+        }.awaitAll().mapNotNull { (provenance, fileList) ->
+            fileList?.let {
```

Comment:
This transformation could have been done asynchronously as well.

Reply:
@oheger-bosch See #2030.